### PR TITLE
Updates ProjectTopicCreate

### PR DESCRIPTION
**Pull request by daniellemaxwell**
### Scope of changes

Updates ProjectTopicCreate so that a CreateTopic request is sent to Ensign and adds changes to corresponding test.

Fixes SC-13771
Fixes SC-13772

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/tenant/topics.go [hunk 1]
```diff
@@ -69,68 +70,101 @@ func (s *Server) ProjectTopicCreate(c *gin.Context) {
...
+	// Retrieve project from the database.
+	var project *db.Project
+	if project, err = db.RetrieveProject(ctx, projectID); err != nil {
+		log.Error().Err(err).Str("projectID", projectID.String()).Msg("could not retrieve project from database")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("project not found"))
 		return
 	}
 
-	c.JSON(http.StatusCreated, t.ToAPI())
+	// Ensure a project cannot be created in a different organization.
```
Comment (bbengfort) [inline pkg/tenant/topics.go:126]:
```suggestion
	// Ensure project belongs to the organization of the requesting user.
```

File: pkg/tenant/topics.go [hunk 2]
```diff
@@ -69,68 +70,101 @@ func (s *Server) ProjectTopicCreate(c *gin.Context) {
...
+
+	var rep *qd.LoginReply
+	if rep, err = s.quarterdeck.ProjectAccess(ctx, req); err != nil {
+		log.Error().Err(err).Msg("could not get access to project claims")
+		c.JSON(qd.ErrorStatus(err), api.ErrorResponse("could not create topic"))
+		return
+	}
+
+	// Create Ensign context.
+	enCtx := qd.ContextWithToken(ctx, rep.AccessToken)
```
Comment (bbengfort) [inline pkg/tenant/topics.go:146]: We'll probably want to create a parallel story for creating a specific ensign context; `qd` I assume, is the quarterdeck library - which might not work the same as Ensign. 

```suggestion
	// Create Ensign context.
	// TODO: ensure the context has PerRPCCredentials for gRPC authentication
	enCtx := qd.ContextWithToken(ctx, rep.AccessToken)
```

File: pkg/tenant/topics.go [hunk 3]
```diff
@@ -69,68 +70,101 @@ func (s *Server) ProjectTopicCreate(c *gin.Context) {
...
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not create topic"))
+		return
+	}
+
+	out := &api.Topic{
+		ID:       string(enTopic.Id),
+		Name:     enTopic.Name,
+		Created:  enTopic.Created.AsTime().Format(time.RFC3339Nano),
+		Modified: enTopic.Modified.AsTime().Format(time.RFC3339Nano),
+	}
```
Comment (bbengfort) [inline pkg/tenant/topics.go:165]: Does any of this need to be saved in the `trtl` database as well?

File: pkg/tenant/topics.go [hunk 4]
```diff
@@ -69,68 +70,101 @@ func (s *Server) ProjectTopicCreate(c *gin.Context) {
...
+	if projectID, err = ulid.Parse(c.Param("projectID")); err != nil {
+		log.Error().Err(err).Str("projectID", projectID.String()).Msg("could not parse project ulid")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("project not found"))
+		return
 	}
 
-	// Add topic to the database and return a 500 response if not successful.
-	if err = db.CreateTopic(c.Request.Context(), t); err != nil {
-		log.Error().Err(err).Msg("could not create project topic in the database")
-		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not add project topic"))
```
Comment (bbengfort) [inline pkg/tenant/topics.go:129]: Is there a reason we deleted saving the metadata to the trtl database?

**Review by bbengfort — COMMENTED** [review]
Looks like a good start! I had a couple of questions to ensure I understood exactly what is happening.

File: pkg/tenant/topics.go (hunk 4 above)
Comment (daniellemaxwell) [reply pkg/tenant/topics.go:129]: I was thinking of returning the data directly from Ensign, but now I realize that this is a skipped step and it should be added back. Adding this back should then save the data in trtl, then what goes out will have the data from trtl. So, I think that answers your other question.

**Comment by bbengfort** [comment]
Thanks - I'll finish the review now!

File: pkg/tenant/topics.go (hunk 4 above)
Comment (bbengfort) [reply pkg/tenant/topics.go:129]: Sorry, I think I jumped ahead - ping me when you're ready for me to review!

**Review by bbengfort — APPROVED** [review]

**Pull request merged**